src/main/java/org/prebid/cache/handlers/cache/PostCacheHandler.java

```diff
@@ -129,8 +130,12 @@ public Mono<ServerResponse> save(final ServerRequest request) {
     }
 
     private boolean isWriteAllowed(final ServerRequest request) {
-        return !apiConfig.isCacheWriteSecured()
-                || StringUtils.equals(request.headers().firstHeader(API_KEY_HEADER), apiConfig.getApiKey());
+        final boolean isValidApiKey = StringUtils.equals(request.headers().firstHeader(API_KEY_HEADER),
+                apiConfig.getApiKey());
+        if (!isValidApiKey) {
+            metricsRecorder.markMeterForTag(metricTagPrefix, MeasurementTag.UNAUTHORIZED_WRITE);
+        }
```

Comment:
Please, move metrics emitting to where current method is used. It will be so confusing to find out at some point that method with naming convention `isSomethingTrue` does side effects.